docs/contracts/inheritance.rst

```diff
@@ -108,93 +115,170 @@ Details are given in the following example.
     // If a constructor takes an argument, it needs to be
     // provided in the header or modifier-invocation-style at
     // the constructor of the derived contract (see below).
-    contract PriceFeed is Owned, Destructible, Named("GoldFeed") {
+    contract PriceFeed is Owned, Pausable, Named("GoldFeed") {
+        uint info;
+
         function updateInfo(uint newInfo) public {
             if (msg.sender == owner) info = newInfo;
         }
 
         // Here, we only specify `override` and not `virtual`.
         // This means that contracts deriving from `PriceFeed`
-        // cannot change the behavior of `destroy` anymore.
-        function destroy() public override(Destructible, Named) { Named.destroy(); }
+        // cannot change the behavior of `pause` anymore.
+        function pause() public override(Pausable, Named) { Named.pause(); }
         function get() public view returns(uint r) { return info; }
-
-        uint info;
     }
 
-Note that above, we call ``Destructible.destroy()`` to "forward" the
-destruction request. The way this is done is problematic, as
+Note that above, we call ``Pausable.pause()`` to "forward" the
+pause request. The way this is done is problematic, as
 seen in the following example:
 
 .. code-block:: solidity
 
     // SPDX-License-Identifier: GPL-3.0
     pragma solidity >=0.7.0 <0.9.0;
-    // This will report a warning due to deprecated selfdestruct
 
-    contract owned {
-        constructor() { owner = payable(msg.sender); }
+    contract Owned {
         address payable owner;
+        constructor() { owner = payable(msg.sender); }
     }
 
-    contract Destructible is owned {
-        function destroy() public virtual {
-            if (msg.sender == owner) selfdestruct(owner);
+    contract Pausable is Owned {
+        bool private _paused;
+
+        event Paused();
+        event Unpaused();
+
+        constructor() { _paused = false; }
+
+        function paused() public view virtual returns (bool) {
+            return _paused;
+        }
+
+        function pause() virtual public {
+            if (msg.sender == owner && !_paused)
+            {
```

Comment:
This seems to be the prevalent style used for braces on this page and it would be best to keep it consistent:
```suggestion
            if (msg.sender == owner && !_paused) {
```

There are also more places like that below.